esmvalcore/preprocessor/_regrid.py

```diff
@@ -433,14 +434,17 @@ def _vertical_interpolate(cube, src_levels, levels, interpolation,
         src_levels.points, cube.shape, cube.coord_dims(src_levels))
 
     # force mask onto data as nan's
-    if np.ma.is_masked(cube.data):
-        cube.data[cube.data.mask] = np.nan
+    if da.ma.getmaskarray(cube.core_data()).any():
+        if cube.has_lazy_data():
+            cube.core_data()[da.ma.getmaskarray(cube.core_data())] = np.nan
+        else:
+            cube.data[cube.data.mask] = np.nan
```

Comment:
Maybe this could be simplified to
```python
cube.data = da.ma.filled(cube.core_data(), np.nan)
```
by using [dask.array.ma.filled](https://docs.dask.org/en/latest/array-api.html#dask.array.ma.filled)?

Reply:
interesting, didn't know about this - I like it! Let me try it out :beer:

Reply:
What I meant is that you can replace the whole block, there should be no need to check if the data is masked first, the function will do that for you, according to its documentation

Reply:
you need to check for the existence of a mask for the realized, non-lazy case because otherwise it will fail with AttributeError, man

Reply:
Can you implement a unit test that shows that? I tried changing it and the unit tests in pass just fine

Reply:
well that's because there is a test I already added. You need :eyeglasses: B-man? :grin:

Reply:
That test passes just fine with the code change I'm proposing, no AttributeError occurs?

Reply:
Bouwe man, can you post the code you wrote here, we may be talking about the same thing but thinking otherwise